### PR TITLE
Make 'PackageCollectionsSigningLibc' target conditional

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
With https://github.com/apple/swift-crypto/pull/94, swift-crypto's
`Crypto` depends on `CCryptoBoringSSL` conditionally, on non-Apple
platforms only.

Since `PackageCollectionsSigningLibc` imports `CCryptoBoringSSL`
through `Crypto`, it fails to compile on Apple platforms.

Modifications:
Define `PackageCollectionsSigningLibc` target and associated
dependencies conditionally on platforms where it's available.

I tried using `.target(name:when:)` in combination with dynamic target
array, but got source not found error for
`PackageCollectionsSigningLibc`.

I also tried just changing target dependency on
`PackageCollectionsSigningLibc` to be conditional, but apparently
`PackageCollectionsSigningLibc` still gets compiled despite not being
used, thus causing build to fail.

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
Looks like infra issue.

@swift-ci please smoke test macOS

File: Package.swift [hunk 1]
```diff
@@ -62,6 +62,50 @@ if useSwiftCryptoV2 {
...
+#else
+packageCollectionSigningTargets = [
+    .target(
+         /** Package collections signing */
+         name: "PackageCollectionsSigning",
+         dependencies: [
+            "Basics",
+            .product(name: "Crypto", package: "swift-crypto"),
+            "PackageCollectionsModel",
+         ],
```
Comment (compnerd) [inline Package.swift:102]: Can we always define the target but just add the dependency conditionally to avoid the duplication?

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]

**Pull request merged**